### ISSUE TITLE
`no-whitespace-within-word` false positive with dashes & I

### ISSUE THREAD

**Issue by mydea** [ember-template-lint/ember-template-lint#2420]
I.e. the following text:

```
Hello - I like this.
```

Triggers an error by the `no-whitespace-within-word` rule. The reason is, that it treats the dash as a word character, I guess.

**Comment by rrenno** [comment]
I think this PR caused a regression for the ability to use digits back-to-back, i.e. 

```javascript
/\d+-\d+/
```

Created an issue here: https://github.com/ember-template-lint/ember-template-lint/issues/2464
